lib/paypal_express/api.rb

```diff
@@ -59,15 +60,27 @@ def purchase_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb
         # Pass extra parameters for the gateway here
         options = {}
 
-        add_required_options(kb_payment_transaction_id, kb_payment_method_id, context, options)
+        if find_value_from_properties(properties, 'from_hpp')
```

Comment:
Ok, I wasn't quite sure from the KB documentation for `buildFormDescriptor` on how it would be used. I'll replace it with a check using the info that `add_required_options`already fetches.